---
title: " Novoalign Testing Report"
author: "Muhamad Amir Hafizin"
date: '`r format(Sys.Date(),"%d %B, %Y")`'
output:
   ioslides_presentation:
html_document:
theme: united
---

# Introduction

*****

* INTRODUCTION
	+ NOVOALIGN = using the genome indexing tools
	+ BWA = using BWT(Burrows-Wheeler Transform)
	+ The table show the the number of mapped region/basepair  based on the each chromosome that is including the PT(plasmid),MT(mitochondria),chr1-5(chromosome 1-5)
	+ Each chromosome was obtained by using the SAMTOOLS from the sorted BAM file file to get each of chromosome by using the command
		+ samtools view in.bam chr1 -b > out.bam (getting the chromosome 1)
	+ The merge of each chromosome from ART and aligner was done by using the perl (perl scriptome)
	+ The comparision was done by merge the SAM file from the ART with the Aligner SAM file by using perl based on the same basepair ID.
	+ The the number lines is find the calculate the number of Basepair ID that. 1 basepair ID = 1 lines
	+ The percent is calculate based on the SAM file from ART (it mapped 100%)
	+ The unmapped from ART meaning the aligner has not been able to mapped that region/basepair to the reference sequence
	+ The different chromosome meaning that the aligner has mapped to the diffrent chromosome than the SAM file from ART (based on the same basepair ID)
	+ The start and stop position for above 50 or below -50 meaning the aligner has mapped outside the acceptable range based on the length of basepair been stimulate from ART(100bp)
	+ The basepair that within the acceptable range meaning the aligner has mapped within the the acceptable range of basepair range based on the length of basepairs (100bp)
	+ The basepair that have start and stop position exectly 0 meaning the aligner has mapped exectly same as ART 

*****

## TABLE FOR EACH CHROMOSOME 
### NOVA LINES (NOVO T-154)
```{r,echo= FALSE, results="asis"}
tab= read.csv("/export/home/amirh/novoalign_testing/bedtool/cigar/bwanovo/novoeachchr.csv",header=TRUE)
library("knitr")
kable(tab)
```


### NOVA PERCENT (NOVO T-154)

```{r,echo= FALSE, results="asis"}
tab= read.csv("/export/home/amirh/novoalign_testing/bedtool/cigar/bwanovo/novoeachchrpercent.csv",header=TRUE)
library("knitr")
kable(tab)
```

-----

### BWA LINES (BWA I-7)

```{r,echo= FALSE, results="asis"}
tab= read.csv("/export/home/amirh/novoalign_testing/bedtool/cigar/bwanovo/bwaeachchr.csv",header=TRUE)
library("knitr")
kable(tab)
```



### BWA PERCENT (BWA I-7)

```{r,echo= FALSE, results="asis"}
tab= read.csv("/export/home/amirh/novoalign_testing/bedtool/cigar/bwanovo/bwaeachchrpercent.csv",header=TRUE)
library("knitr")
kable(tab)
```

-----

### Comparision between NOVO and BWA In percentage based on full Genome

```{r,echo= FALSE, results="asis"}
tab= read.csv("/export/home/amirh/novoalign_testing/bedtool/cigar/bwanovo/percent.csv",header=TRUE)
library("knitr")
kable(tab)
```


----


### Comparision between NOVO(N) and BWA(B) in percentage based on each Chromosome

```{r,echo= FALSE, results="asis"}
tab= read.csv("/export/home/amirh/novoalign_testing/bedtool/cigar/bwanovo/chrpercent.csv",header=TRUE)
library("knitr")
kable(tab)
```

----

### Comparision between BWA and multiple Novo in number of Lines

```{r,echo= FALSE, results="asis"}
tab= read.csv("/export/home/amirh/novoalign_testing/bedtool/cigar/bwanovo/all.csv",header=FALSE)
library("knitr")
kable(tab)
```

*****

### Comparision between BWA and multiple Novo (Percent)

```{r,echo= FALSE, results="asis"}
tab= read.csv("/export/home/amirh/novoalign_testing/bedtool/cigar/bwanovo/allpercent.csv",header=FALSE)
library("knitr")
kable(tab)
```

-----

### Comparision between BWA and multiple Novo Full (Percent)

```{r,echo= FALSE, results="asis"}
tab= read.csv("/export/home/amirh/novoalign_testing/bedtool/cigar/bwanovo/full.csv",header=TRUE)
library("knitr")
kable(tab)
```


```{r, echo= FALSE, results="asis"}
x<-read.csv("/export/home/amirh/novoalign_testing/bedtool/cigar/bwanovo/fullg.csv",sep=",",header=TRUE,stringsAsFactors = TRUE)
d<-t(x)
barplot(d[,1],main=" MAPPED FULL (percent) VS Threshold/missmatch",ylab="Percent Full",xlab="Threshold/missmatch",border="red",ylim=c(0,100))
```

*****

### Comparision between BWA and multiple Novo PT (Percent)

```{r,echo= FALSE, results="asis"}
tab= read.csv("/export/home/amirh/novoalign_testing/bedtool/cigar/bwanovo/pt.csv",header=TRUE)
library("knitr")
kable(tab)
```


```{r, echo= FALSE, results="asis"}
x<-read.csv("/export/home/amirh/novoalign_testing/bedtool/cigar/bwanovo/ptg.csv",sep=",",header=TRUE,stringsAsFactors = TRUE)
d<-t(x)
barplot(d[,1],main=" MAPPED PT (percent) VS Threshold/missmatch",ylab="Percent PT",xlab="Threshold/missmatch",border="red",ylim=c(0,100))
```

*****

### Comparision between BWA and multiple Novo MT (Percent)

```{r,echo= FALSE, results="asis"}
tab= read.csv("/export/home/amirh/novoalign_testing/bedtool/cigar/bwanovo/mt.csv",header=TRUE)
library("knitr")
kable(tab)
```


```{r, echo= FALSE, results="asis"}
x<-read.csv("/export/home/amirh/novoalign_testing/bedtool/cigar/bwanovo/mtg.csv",sep=",",header=TRUE,stringsAsFactors = TRUE)
d<-t(x)
barplot(d[,1],main=" MAPPED MT (percent) VS Threshold/missmatch",ylab="Percent MT",xlab="Threshold/missmatch",border="red",ylim=c(0,100))
```

*****

### Comparision between BWA and multiple Novo CHR1 (Percent)

```{r,echo= FALSE, results="asis"}
tab= read.csv("/export/home/amirh/novoalign_testing/bedtool/cigar/bwanovo/chr1.csv",header=TRUE)
library("knitr")
kable(tab)
```


```{r, echo= FALSE, results="asis"}
x<-read.csv("/export/home/amirh/novoalign_testing/bedtool/cigar/bwanovo/chr1g.csv",sep=",",header=TRUE,stringsAsFactors = TRUE)
d<-t(x)
barplot(d[,1],main=" MAPPED CHR1 (percent) VS Threshold/missmatch",ylab="Percent CHR1",xlab="Threshold/missmatch",border="red",ylim=c(0,100))
```

*****

### Comparision between BWA and multiple Novo CHR2 (Percent)

```{r,echo= FALSE, results="asis"}
tab= read.csv("/export/home/amirh/novoalign_testing/bedtool/cigar/bwanovo/chr2.csv",header=TRUE)
library("knitr")
kable(tab)
```


```{r, echo= FALSE, results="asis"}
x<-read.csv("/export/home/amirh/novoalign_testing/bedtool/cigar/bwanovo/chr2g.csv",sep=",",header=TRUE,stringsAsFactors = TRUE)
d<-t(x)
barplot(d[,1],main=" MAPPED CHR2 (percent) VS Threshold/missmatch",ylab="Percent CHR2",xlab="Threshold/missmatch",border="red",ylim=c(0,100))
```

*****

### Comparision between BWA and multiple Novo CHR3 (Percent)

```{r,echo= FALSE, results="asis"}
tab= read.csv("/export/home/amirh/novoalign_testing/bedtool/cigar/bwanovo/chr3.csv",header=TRUE)
library("knitr")
kable(tab)
```


```{r, echo= FALSE, results="asis"}
x<-read.csv("/export/home/amirh/novoalign_testing/bedtool/cigar/bwanovo/chr3g.csv",sep=",",header=TRUE,stringsAsFactors = TRUE)
d<-t(x)
barplot(d[,1],main=" MAPPED CHR3 (percent) VS Threshold/missmatch",ylab="Percent CHR3",xlab="Threshold/missmatch",border="red",ylim=c(0,100))
```

*****

### Comparision between BWA and multiple Novo CHR4 (Percent)

```{r,echo= FALSE, results="asis"}
tab= read.csv("/export/home/amirh/novoalign_testing/bedtool/cigar/bwanovo/chr4.csv",header=TRUE)
library("knitr")
kable(tab)
```


```{r, echo= FALSE, results="asis"}
x<-read.csv("/export/home/amirh/novoalign_testing/bedtool/cigar/bwanovo/chr4g.csv",sep=",",header=TRUE,stringsAsFactors = TRUE)
d<-t(x)
barplot(d[,1],main=" MAPPED CHR4 (percent) VS Threshold/missmatch",ylab="Percent CHR4",xlab="Threshold/missmatch",border="red",ylim=c(0,100))
```

*****

### Comparision between BWA and multiple Novo CHR5 (Percent)

```{r,echo= FALSE, results="asis"}
tab= read.csv("/export/home/amirh/novoalign_testing/bedtool/cigar/bwanovo/chr5.csv",header=TRUE)
library("knitr")
kable(tab)
```

*****



```{r, echo= FALSE, results="asis"}
x<-read.csv("/export/home/amirh/novoalign_testing/bedtool/cigar/bwanovo/chr5g.csv",sep=",",header=TRUE,stringsAsFactors = TRUE)
d<-t(x)
barplot(d[,1],main=" MAPPED CHR5 (percent) VS Threshold/missmatch",ylab="Percent ",xlab="Threshold/missmatch",border="red",ylim=c(0,100))
```

*****

### VennDiagram
##### the acceptable basepair location based on the merge with ART (No of LINES)

```{r,echo= FALSE, results="asis"}
library("VennDiagram")
venn.plot <- draw.pairwise.venn(11676041,11672372,11576099,c("novo","BWA"),scaled=FALSE)
```

*****

##### the percentage of the accetable basepair location

```{r,echo= FALSE, results="asis"}
library("VennDiagram")
venn.plot <- draw.pairwise.venn(97.73,97.70,96.9,c("novo","BWA"),scaled=FALSE)
```

*****


##### the acceptable basepair location based on the merge with ART (No of LINES)

```{r,echo= FALSE, results="asis"}
library("VennDiagram")
venn.plot <- draw.pairwise.venn(11287984,11672191,11191893,c("novo","BWA"),scaled=FALSE)
```

******


##### the percentage of the accetable basepair location

```{r,echo= FALSE, results="asis"}
library("VennDiagram")
venn.plot <- draw.pairwise.venn(94.48,97.70,93.68,c("novo","BWA"),scaled=FALSE)
```




